agent/eventhandler/task_handler_types.go

```diff
@@ -207,6 +204,12 @@ func setTaskChangeSent(event *sendableEvent, dataClient data.Client) {
 		container := containerStateChange.Container
 		updateContainerSentStatus(container, containerStateChange.Status, dataClient)
 	}
+	for _, managedAgentStateChange := range event.taskChange.ManagedAgents {
+		container := managedAgentStateChange.Container
```

Comment:
nit: this and below variable can be avoided, if we just pass the fields from the `managedAgentStateChange` struct